lib/chef/resource/windows_share.rb

```diff
@@ -336,8 +331,28 @@ def bool_string(bool)
           # bool ? 1 : 0
           bool ? "$true" : "$false"
         end
+
+        # Grant-SmbShareAccess sets only one permission to one user so we need to remove from the lower permission access
+        # For change_users remove common full_users from it
+        # For read_users remove common full_users as well as change_users
+        def set_users
+          new_resource.change_users = new_resource.change_users - new_resource.full_users
```

Comment:
Modifying new_resource properties mid run is something we generally don't recommend. This needs to be handled in a new variable.

Reply:
Hi @tas50 , I have tried by using new variable, still it will fail to acheive idempotency, I have also tried by another approach(re arranging the current_resource) but it will require lot of changes in the code, so any suggestion for the solution?

Reply:
i think the problem is that "change_users" is a verb-style property rather than a noun, which breaks the simple idempotency model, which means converge_if_changed cannot be used, since that strictly compares a desired state with a current state.

most likely abandoning converge_if_changed is the right thing to do here, this resource is outside of the scope of what it can handle.